core/bin/qdrant_migrator.rs

```diff
@@ -0,0 +1,434 @@
+use std::str::FromStr;
+
+use anyhow::{anyhow, Result};
+use clap::{Parser, Subcommand};
+use dust::{
+    data_sources::qdrant::{QdrantClients, QdrantCluster, QdrantDataSourceConfig},
+    project,
+    run::Credentials,
+    stores::postgres,
+    stores::store,
+    utils,
+};
+use qdrant_client::{
+    prelude::Payload,
+    qdrant::{self, PointId, ScrollPoints},
+};
+
+#[derive(Debug, Subcommand)]
+enum Commands {
+    #[command(arg_required_else_help = true)]
+    #[command(about = "Show qdrant state for data source", long_about = None)]
+    Show {
+        project_id: i64,
+        data_source_id: String,
+    },
+    #[command(arg_required_else_help = true)]
+    #[command(about = "Set `shadow_write_cluster` \
+                       (!!! creates collection on `shadow_write_cluster`)", long_about = None)]
+    SetShadowWrite {
+        project_id: i64,
+        data_source_id: String,
+        cluster: String,
+    },
+    #[command(arg_required_else_help = true)]
+    #[command(about = "Clear `shadow_write_cluster` \
+                       (!!! deletes collection from `shadow_write_cluster`)", long_about = None)]
+    ClearShadowWrite {
+        project_id: i64,
+        data_source_id: String,
+    },
+    #[command(arg_required_else_help = true)]
+    #[command(about = "Migrate `cluster` collection to `shadow_write_cluster`", long_about = None)]
+    MigrateShadowWrite {
+        project_id: i64,
+        data_source_id: String,
+    },
+    #[command(arg_required_else_help = true)]
+    #[command(about = "Switch `shadow_write_cluster` and `cluster` \
+                       (!!! moves read traffic to `shadow_write_cluster`)", long_about = None)]
+    CommitShadowWrite {
+        project_id: i64,
+        data_source_id: String,
+    },
+}
+
+/// A fictional versioning CLI
```

Comment:
Not sure I understand what is fictional about it, wdym ?

Reply:
cargo cult from a lib

Reply:
lol